rewrite-gradle/src/test/java/org/openrewrite/gradle/marker/GradleProjectTest.java

```diff
@@ -76,6 +76,48 @@ void noopUpgrade() {
         );
     }
 
+    @Test
+    void repositoryWithBasicCredentials() {
+        rewriteRun(
+          spec -> spec.recipe(new UpgradeDependencyInMarker(
+            new GroupArtifactVersion("org.openrewrite", "rewrite-java", "8.56.0"),
+            "implementation",
+            (original, updated) -> assertThat(updated)
+              .isSameAs(original)
+              .extracting(GradleProject::getMavenRepositories)
+              .satisfies(list -> assertThat(list)
+                .singleElement()
+                .satisfies(repo -> {
+                      assertThat(repo.getUri()).isEqualTo("https://example.com/maven2");
+                      assertThat(repo.getUsername()).isEqualTo("dummyuser");
+                      assertThat(repo.getPassword()).isEqualTo("dummypass");
```

Comment:
Somehow these assertions still fail locally; figured it'd be good to verify these are picked up by the builder, but I'm not sure if the builder is even exercised for this test path.

Reply:
Fixed via https://github.com/openrewrite/rewrite/commit/aa9c44d7cd6b036bf1a80167a183f62fe667b03b